provider/bikeshed/src/list/mod.rs

```diff
@@ -53,26 +53,27 @@ fn load<M: DataMarker<Yokeable = ListFormatterPatternsV1<'static>>>(
     ])?;
 
     if langid.language == language!("es") {
-        if M::INFO == AndListV1Marker::INFO || M::INFO == UnitListV1Marker::INFO {
+        if M::INFO == AndListV2Marker::INFO || M::INFO == UnitListV2Marker::INFO {
             static I_SOUND: OnceLock<SerdeDFA<'static>> = OnceLock::new();
 
             // Starts with i or (hi but not hia/hie)
             let i_sound = I_SOUND.get_or_init(|| {
-                SerdeDFA::new(Cow::Borrowed("i|hi([^ae]|$)")).expect("Valid regex")
+                SerdeDFA::new(Cow::Borrowed("^[iI]|(?:[hH][iI](?:[^aeAE]|$))"))
+                    .expect("Valid regex")
```

Comment:
Issue: Why are you changing the regex in the same PR? If you want to change it, seems like that should be a separate PR we can discuss in more detail. It seems like this problem could be solved via other means, for example.

Reply:
0.4 requires an extra feature to support case-insensitive, and I was hoping I could get some size regression back like this. It's done now, I'd rather not redo all of this.